src/main/java/bgu/spl/mics/application/passiveObjects/Squad.java

```diff
@@ -1,87 +1,129 @@
-		if (instance == null)
-			instance = new Squad();
-		return instance;
-	}
+    /**
+     * Retrieves the single instance of this class.
+     *
+     * @pre none
+     * @post return != null
+     */
+    public static Squad getInstance() {
+        if (_instance == null)
+            _instance = new Squad();
+        return _instance;
+    }
 
-	private Squad() {
-		// squad is accessed by Moneypenny, which my have several instances, so thread-safe structure is needed.
-		agents = new ConcurrentHashMap<String, Agent>();
-	}
-	/**
-	 * Initializes the squad. This method adds all the agents to the squad.
-	 * <p>
-	 * @param agents 	Data structure containing all data necessary for initialization
-	 * 						of the squad.
-	 * @pre none
-	 * @post (for agent : agents {getAgents(agent.getSerialNumber) == true})
-	 */
-	public void load (Agent[] agents) {
-		// TODO Implement this
-	}
+    private Squad() {
+        // squad is accessed by Moneypenny, which my have several instances, so thread-safe structure is needed.
+        _agents = new ConcurrentHashMap<>();
+    }
 
-	/**
-	 * Releases agents.
-	 * @pre (for serial : serials {getAgents([serial]).state == BLOCKED})
-	 * @post (getAgents(serials) == true)
-	 */
-	public void releaseAgents(List<String> serials){
-		// TODO Implement this
-	}
+    /**
+     * Initializes the squad. This method adds all the agents to the squad.
+     * <p>
+     *
+     * @param agents Data structure containing all data necessary for initialization of the squad.
+     *
+     * @pre none
+     * @post (for agent : agents { getAgents ( agent.getSerialNumber) == true})
+     */
+    public void load(Agent[] agents) {
+        for (Agent agent : agents) {
+            this._agents.put(agent.getSerialNumber(), agent);
+        }
+    }
 
-	/**
-	 * simulates executing a mission by calling sleep.
-	 * @param time   milliseconds to sleep
-	 * @pre (for serial : serials {getAgents([serial]).state == BLOCKED})
-	 * @post (current_time >= start_time + time && getAgents(serials) == true)
-	 */
-	public void sendAgents(List<String> serials, int time){
-		// TODO Implement this
-	}
+    /**
+     * Releases agents.
+     *
+     * @pre (for serial : serials { getAgents ( [serial]).state == BLOCKED})
+     * @post (getAgents ( serials) == true)
+     */
+    public void releaseAgents(List<String> serials) {
+        Agent agent;
+    	for (String serial : serials) {
+            agent = _agents.get(serial);
+            agent.release();
+            agent.notifyAll();
+        }
+    }
+
+    /**
+     * simulates executing a mission by calling sleep.
+     *
+     * @param time milliseconds to sleep
+     *
+     * @pre (for serial : serials { getAgents ( [serial]).state == BLOCKED})
+     * @post (current_time > = start_time + time & & getAgents ( serials) == true)
+     */
+    public void sendAgents(List<String> serials, int time) {
+        try {Thread.sleep(time);} catch (InterruptedException e) {}
```

Comment:
This should wait in time ticks. I have no Idea how...
Maybe save the current time tick before the sleep and save an object that saves the current time tick in a local variable, by subscribing to the TimeTick broadcast

Reply:
yes, it seems the only way, but squad should be, as stated, Passive object!!